api/server/cluster.go

```diff
@@ -20,10 +21,18 @@ const (
 
 type clusterApi struct {
 	restBase
+	SecretManager *secrets.Manager
 }
 
-func newClusterAPI() restServer {
-	return &clusterApi{restBase{version: cluster.APIVersion, name: "Cluster API"}}
+type ClusterServerConfiguration struct {
+	ConfigSecretManager *secrets.Manager
+}
+
+func newClusterAPI(config ClusterServerConfiguration) restServer {
+	return &clusterApi{restBase: restBase{version: cluster.APIVersion,
+		name: "Cluster API"},
+		SecretManager: config.ConfigSecretManager,
```

Comment:
Please format correctly:

```go
return &clusterApi{
    restBase: restBase{
        version: cluster.APIVersion,
	name: "Cluster API"
    },
    SecretManager: config.ConfigSecretManager,
}
```

Reply:
ok, will do. thanks

Reply:
also unit test failure is occured since signature of newClusterAPI() is changed, will do necessary changes with help of imran on this